resource_leak.py

```diff
@@ -0,0 +1,44 @@
+# DB resource leak demo
+import psycopg2
+
+def test_connection():
+    connection = psycopg2.connect(user="postgres",
+                                  password="postgres123",
+                                  host="127.0.0.1",
+                                  port="5432",
+                                  database="postgres_db")
+    cursor = connection.cursor()
+    cursor.execute("SELECT version();")
+    record = cursor.fetchone()
+
+# File resource leak demo
+def read_write_data(data):
+    file = open("test-file.txt", "r")
+    out = file.read()
+    file.write(out)
+
+
+def read_lines(file):
+    lines = []
+    f = open(file, "r")
+    for line in f:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)